src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -23,7 +23,7 @@ export class SavedObjectsClient {
    * @returns {promise} - { id, type, version, attributes }
   */
   async create(type, attributes = {}, options = {}) {
-    const method = options.overwrite ? 'index' : 'create';
+    const method = get(options, 'overwrite', false) === false && options.id ? 'create' : 'index';
```

Comment:
`index` handles automatic id creation, in addition, it allows for overwriting of a document when an ID is specified.